src/mgr/ActivePyModules.cc

```diff
@@ -467,6 +467,24 @@ void ActivePyModules::start_one(PyModuleRef py_module)
   }));
 }
 
+void ActivePyModules::stop()
+{
+  std::lock_guard locker(lock);
+
+  for (auto& [name, module] : modules) {
+    lock.unlock();
```

Comment:
I wonder if this unlock is unsafe. Look at `notify_all` which uses a finisher to complete the call. I am concerned that `modules` may be modified during this loop causing the internal iterator of this loop to invalidate.

Reply:
Ummm... probably -- let me check.

Reply:
You may be right -- but FWIW, the existing `shutdown()` call does it this way. There isn't much of a difference between `stop` and `shutdown`.

Reply:
That's correct, and `shutdown` has always been that way since John wrote it. I guess we'll keep it this way until we see a problem...